presto-spi/src/main/java/com/facebook/presto/spi/security/AccessControlContext.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.security;
+
+import com.facebook.presto.spi.QueryId;
+
+import java.util.Optional;
+
+public class AccessControlContext
+{
+    private final QueryId queryId;
+    private final Optional<String> clientInfo;
+
+    public AccessControlContext(QueryId queryId, Optional<String> clientInfo)
+    {
+        this.queryId = queryId;
+        this.clientInfo = clientInfo;
+    }
+
+    public String getQueryId()
+    {
+        return queryId.getId();
+    }
+
+    public String getClientInfo()
```

Comment:
just return `Optional<String>` that way the implementation can decide what to do if the value is not present.